source/frontend/src/features/properties/map/compensation/CompensationRequisitionTrayView.tsx

```diff
@@ -0,0 +1,129 @@
+import clsx from 'classnames';
+import * as Styled from 'components/common/styles';
+import LoadingBackdrop from 'components/maps/leaflet/LoadingBackdrop/LoadingBackdrop';
+import { Api_AcquisitionFile } from 'models/api/AcquisitionFile';
+import { Api_Compensation } from 'models/api/Compensation';
+import { useState } from 'react';
+import React from 'react';
+import { MdClose } from 'react-icons/md';
+import ReactVisibilitySensor from 'react-visibility-sensor';
+import styled from 'styled-components';
+
+import { CompensationRequisitionDetailContainer } from './detail/CompensationRequisitionDetailContainer';
+import CompensationRequisitionDetailView from './detail/CompensationRequisitionDetailView';
+import UpdateCompensationRequisitionContainer from './update/UpdateCompensationRequisitionContainer';
+import UpdateCompensationRequisitionForm from './update/UpdateCompensationRequisitionForm';
+
+export interface CompensationRequisitionTrayViewProps {
+  compensation?: Api_Compensation;
+  acquistionFile: Api_AcquisitionFile;
+  clientConstant: string;
+  gstConstant: number | undefined;
+  onClose: () => void;
+  loading: boolean;
+  error: boolean;
+  editMode: boolean;
+  setEditMode: (editMode: boolean) => void;
+  onUpdate: () => void;
+}
+
+export const CompensationRequisitionTrayView: React.FunctionComponent<
+  React.PropsWithChildren<CompensationRequisitionTrayViewProps>
+> = ({
+  compensation,
+  acquistionFile,
+  clientConstant,
+  gstConstant,
+  editMode,
+  setEditMode,
+  loading,
+  error,
+  onClose,
+  onUpdate,
+}) => {
+  const [show, setShow] = useState(true);
+
+  let detailViewContent =
+    !editMode && compensation ? (
+      <HalfHeightDiv>
+        {!!compensation?.id && (
+          <CompensationRequisitionDetailContainer
+            compensation={compensation}
+            acquisitionFile={acquistionFile}
+            View={CompensationRequisitionDetailView}
+            clientConstant={clientConstant}
+            gstConstant={gstConstant ?? 0}
+            loading={false}
+            setEditMode={setEditMode}
+          ></CompensationRequisitionDetailContainer>
+        )}
+      </HalfHeightDiv>
+    ) : undefined;
+
+  let updateViewConent =
+    editMode && compensation ? (
+      <HalfHeightDiv>
+        <UpdateCompensationRequisitionContainer
+          compensation={compensation}
+          acquisitionFile={acquistionFile}
+          formikRef={React.createRef()}
+          onSuccess={() => {
+            //setShow(false);
```

Comment:
do not leave commented out code unless there is a good reason.

Reply:
updated.